### ISSUE TITLE
ExtractSignatures Error

### ISSUE THREAD

**Issue by wjaratlerdsiri** [PoisonAlien/maftools#8]
Dear Author,

I used MAF file from Oncotator. Every step was fine, including 'TrinucleotideMatrix'. But, when I try 'extractSignatures', I saw the following error. Could you help me with this? Thankyou

>AP1.hg19.tnm = trinucleotideMatrix(maf = AP1, ref_genome = '/Users/hg19_chromosome.fa', prefix = 'chr', add = TRUE, ignoreChr = 'chrM', useSyn = TRUE)
> AP1.hg19.sign = extractSignatures(mat = AP1.hg19.tnm)
Estimating best rank..
Timing stopped at: 0.002 0 0.002 
Timing stopped at: 0.002 0 0.002 
Timing stopped at: 0.001 0 0.001 
Timing stopped at: 0.002 0 0.003 
Timing stopped at: 0.002 0 0.003 
Error in (function (...)  : All the runs produced an error:
	-#1 [r=2] -> none of the packages are loaded [in call to 'path.package']
	-#2 [r=3] -> none of the packages are loaded [in call to 'path.package']
	-#3 [r=4] -> none of the packages are loaded [in call to 'path.package']
	-#4 [r=5] -> none of the packages are loaded [in call to 'path.package']
	-#5 [r=6] -> none of the packages are loaded [in call to 'path.package']

Cheers,
James

**Comment by PoisonAlien** [comment]
Hi,

this is a known issue, for some reason nmf function loaded via namespace gives this error.

try manually loading package before running extractSignatures. It should work fine.

```
library('NMF')
AP1.hg19.tnm = trinucleotideMatrix(maf = AP1, ref_genome = '/Users/hg19_chromosome.fa', prefix = 'chr', add = TRUE, ignoreChr = 'chrM', useSyn = TRUE)
```

**Comment by wjaratlerdsiri** [comment]
Thanks James

**Comment by vyellapa** [comment]
Hi James,
Im getting a similar error even after loading the NMF library. Is there is any other obvious fix to this?

Thanks,
Teja

> library('NMF')
> laml.sign = extractSignatures(mat = laml.tnm)
Warning : Found zero mutations for conversions T[T>G]C
Estimating best rank..
Error in (function (...)  : All the runs produced an error:
	-#1 [r=2] -> attempt to set 'rownames' on an object with no dimensions [in call to 'rownames<-']
	-#2 [r=3] -> attempt to set 'rownames' on an object with no dimensions [in call to 'rownames<-']
	-#3 [r=4] -> attempt to set 'rownames' on an object with no dimensions [in call to 'rownames<-']
	-#4 [r=5] -> attempt to set 'rownames' on an object with no dimensions [in call to 'rownames<-']
	-#5 [r=6] -> attempt to set 'rownames' on an object with no dimensions [in call to 'rownames<-']

**Comment by wjaratlerdsiri** [comment]
I worked out on this by specifying exact values of n. I think because my mutation data were not large enough to run the default n parameter and find the best signature.

So I ran n=1 to n=5 separately and chose signature with the highest correlation as described in the manual. I noticed that the same error came up again when you used large n.

My command:
library('NMF')
install.extras('NMF')

A = trinucleotideMatrix(maf = AP, ref_genome = 'hg19_chromosome.fa', prefix = 'chr', add = TRUE, ignoreChr = 'chrM', useSyn = TRUE)

A2 = extractSignatures(mat = A, n = 1, plotBestFitRes = T)
A2 = extractSignatures(mat = A, n = 2, plotBestFitRes = T)
...
A2 = extractSignatures(mat = A, n = 6, plotBestFitRes = T)

plotSignatures(A2)
